detection_rules/utils.py

```diff
@@ -265,6 +267,26 @@ def normalize_timing_and_sort(events, timestamp='@timestamp', asc=True):
     return event_sort(events, timestamp=timestamp, asc=asc)
 
 
+def normalize_kql_keywords(query: str) -> str:
+    """Normalize KQL keywords to lowercase."""
+
+    def lower_keywords_in_tree(tree: Tree, query_chars: List[str]) -> None:
+        """Process the tree and replace keyword tokens with their lowercase version."""
+        for child in tree.children:
+            if isinstance(child, Token) and child.type == child.value:
+                start = child.start_pos
+                end = start + len(child.value)
+                query_chars[start:end] = child.value.lower()
+            elif isinstance(child, Tree):
+                lower_keywords_in_tree(child, query_chars)
+
+    query_chars = list(query)
+    tree = kql.lark_parse(query)
```

Comment:
Do we really have to parse a second time (independent of the `parse` function?

Reply:
We do not strictly need to parse a second time. This function is currently unused and only exists to satisfy the request to have https://github.com/elastic/detection-rules/issues/3548#issuecomment-2030529220. If we do not want to provide this in this PR we can remove this code.

Reply:
Removed :+1: